managed/services/telemetry/telemetry.go

```diff
@@ -233,10 +233,20 @@ func (s *Service) prepareReport(ctx context.Context) *pmmv1.ServerMetric {
 			continue
 		}
 
-		for _, each := range metrics {
-			telemetryMetric.Metrics = append(telemetryMetric.Metrics, each...)
+		if telemetry.Transform != nil {
+			if telemetry.Transform.Type == JSONTransformType {
+				metrics, err = transformToJSON(&telemetry, metrics)
+				if err != nil {
+					s.l.Debugf("failed to transform to JSON: %s", err)
+					continue
+				}
+			}
```

Comment:
else "undefined transform type %s"?

Reply:
good catch

fixed

Reply:
I'd reword `undefined`. `unsupported` is a more correct word here